java/java-psi-impl/src/com/intellij/lang/java/parser/StatementParser.java

```diff
@@ -160,7 +160,14 @@ else if (tokenType == JavaTokenType.IDENTIFIER || tokenType == JavaTokenType.AT)
       skipQualifiedName(builder);
       IElementType suspectedLT = builder.getTokenType(), next = builder.lookAhead(1);
       refPos.rollbackTo();
-      if (suspectedLT == JavaTokenType.LT || suspectedLT == JavaTokenType.DOT && next == JavaTokenType.AT) {
+
+      // skip the the type with possible type parameters to see if it is followed by "::"
+      myParser.getReferenceParser().parseType(builder, 0);
+      final boolean constructorRef = builder.getTokenType() == JavaTokenType.DOUBLE_COLON;
+
+      refPos.rollbackTo();
```

Comment:
After the first rollback (at line 162), the marker becomes invalid; you need to mark the position again.

Reply:
I moved the check if the examined line is a method reference to a constructor inside of the `if`-statement below. This marker is no longer required